src/SonarScanner.MSBuild.Common/Runtime.cs

```diff
@@ -0,0 +1,29 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2025 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarScanner.MSBuild.Common;
+
+public interface IRuntime
```

Comment:
(Filenames don't match: `Runtime.cs`).

Adding the ILogger interface here is a bit questionable but okay.

Have you considered adding the interface methods here instead?
```cs
public interface IRuntime
    string ReadAllText(string path);
    void WriteAllText(string path, string contents);

    void CreateDirectory(string path);

    string FolderPath(Environment.SpecialFolder folder, Environment.SpecialFolderOption option);

    // ..
```

I think that would have been more consequent. Now we only have one abstraction more on top of abstractions.